src/react-components/src/components/01-atoms/Forms/FormDropdown.tsx

```diff
@@ -16,30 +18,39 @@ export interface FormDropdownProps {
 }
 
 export default function FormDropdown(props: FormDropdownProps) {
-  const { dropdownId, blockName, labelText, options, ariaLabel, disabled, selectedOption, onSelectBlur, onSelectChange } = props;
-  if (!labelText && !ariaLabel) {
-      // TODO: Assign aria-labeledBy to labelText
-      throw new Error("Either labelText or ariaLabel must be defined");
+  const { dropdownId, blockName, options, labelId, isRequired, ariaLabel, disabled, selectedOption, onSelectBlur, onSelectChange } = props;
+  const modifiers = props.modifiers ? props.modifiers : [];
+  if (!labelId && !ariaLabel) {
+    throw new Error("Must either have labelId or aria-label");
   }
 
   let formItemBlockName = blockName ? blockName : "form-item";
-  let labelClassName = bem("label", ["textField"], formItemBlockName);
+
+  let selectProps = {
+    id: dropdownId,
+    className: bem("select", modifiers, formItemBlockName),
+    value: selectedOption,
+    onChange: onSelectChange,
+    onBlur: onSelectBlur,
+    "aria-required": isRequired,
+    disabled: disabled ? disabled : false
+  };
+
+  if (labelId) {
+    selectProps["aria-labelledby"] = labelId;
+  } else {
+    selectProps["aria-label"] = ariaLabel;
+  }
 
   return (
-    <div className={bem("dropdown", [], formItemBlockName)}>
-      {labelText && (<label htmlFor={dropdownId} className={labelClassName}>{labelText}</label>)}
-
-      <select id={dropdownId}
-        className={bem("select", [], formItemBlockName)}
-        value={selectedOption}
-        onChange={onSelectChange}
-        onBlur={onSelectBlur}
-        aria-label={ariaLabel}
-        disabled={disabled ? disabled : false}
-      >
-          { options.map((child, key) => {
-            return <option key={key.toString()} aria-selected={child === selectedOption} value={child}>{ child }</option>;
-          }) }
+    <div className={bem("dropdown", modifiers, formItemBlockName)}>
+
+      <select {...selectProps}>
+        {options.map((child, key) => {
```

Comment:
I would say that there should be a check before the `return` where if `options` is empty, then don't render anything. But, you do have that test where `Pagination` renders everything even if the total pages is 0. I'm not sure of the acceptance criteria for this, but I wouldn't allow components to render anything if they don't have to.

Reply:
>  I wouldn't allow components to render anything if they don't have to. 
I like this principle ... It does both the thing that I want (don't break peoples pages if they make boo-boos) and the thing that you want (don't display stuff that doesn't make sense).  I should ... maybe ... use this elsewhere too.